lib/Utils/SourceNormalization.cpp

```diff
@@ -220,11 +221,19 @@ class MinimalPPLexer: public Lexer {
 
       } else if (First.equals("static") || First.equals("constexpr") ||
                  First.equals("inline") || First.equals("const")) {
+        // First check if the current keyword is "unsigned".
+        llvm::StringRef Modifier = Identifier(Tok);
+        if (Modifier.equals("signed") || Modifier.equals("unsigned"))
+          SeenSignedness = true;
+
         // Advance past keyword for below
         if (!LexClean(Tok))
           return kNONE;
+      } else if (First.equals("signed") || First.equals("unsigned")) {
+        SeenSignedness = true;
       }
 
+SkipIdentifier:
```

Comment:
Any chance we can avoid the `goto`?

Reply:
I first considered a structure like
```c
while (true) {
  // ...
  if (SeenSignedness) {
    continue;
  }
  break;
}
```
but went with `goto` because the infinite loop is just hiding the intent. I'll try refactoring the code into a function, that should work equally well.

Reply:
`SkipIdentifier` is now a function which is potentially called twice.

Alternatively I've tried `AdvanceTo(Tok, tok::l_paren)` instead of the whole logic now in `SkipIdentifier`. Unfortunately this also promotes lambdas or anything else with a brace in it to top level declarator which is probably not wanted.